Add SeamHttpRequest (#67)
seamapi/javascript-http: README.md

An Axios compatible client may be provided to create a `SeamHttp` instance.
This API is used internally and is not directly supported.

#### Inspecting the Request

It is possible to inspect a request before it is sent. All of the client methods return an instance of `SeamHttpRequest` which has `url`, `method` and `data` properties. The request is only sent to the server when the `SeamHttpRequest` is awaited.

```ts
const seam = new SeamHttp('your-api-key')

const request = seam.devices.list()

console.log('Sending request', request.url, request.method, request.data)

const devices = await request // or `await request.execute()` if you prefer

console.log(devices)
```

## Development and Testing

### Quickstart

```

> 

Reviewer: What is going on with these random changes to other parts of the file 🤔

An Axios compatible client may be provided to create a `SeamHttp` instance.
This API is used internally and is not directly supported.

#### Inspecting the Request

All client methods return an instance of `SeamHttpRequest`.

> Inspect the request before it is sent to the server by intentionally not awaiting the `SeamHttpRequest`;

Reviewer: Shift key fail 🤦🏻 
```suggestion
Inspect the request before it is sent to the server by intentionally not awaiting the `SeamHttpRequest`:
```

An Axios compatible client may be provided to create a `SeamHttp` instance.
This API is used internally and is not directly supported.

#### Inspecting the Request

> It is possible to inspect a request before it is sent. All of the client methods return an instance of `SeamHttpRequest` which has `url`, `method` and `body` properties. The request is only sent to the server when the `SeamHttpRequest` is awaited.
> 
> ```ts
> const seam = new SeamHttp('your-api-key')
> 
> const request = seam.devices.list()
> 
> console.log('Sending request', request.url, request.method, request.body)
> 
> const devices = await request // or `await request.execute()` if you prefer
> 
> console.log(devices)
> ```

Reviewer: ```suggestion
All client methods return an instance of `SeamHttpRequest`. 
Inspect the request before it is sent to the server by intentionally not awaiting the `SeamHttpRequest`:

```ts
const seam = new SeamHttp('your-api-key')

const request = seam.devices.list()

console.log(`${request.method} ${request.path}, JSON.stringify(request.body)}`

const devices = await request.execute()
```
